src/main/scala/nodes/learning/KernelGenerator.scala

```diff
@@ -0,0 +1,273 @@
+package nodes.learning
+
+import breeze.linalg._
+import breeze.math._
+import breeze.numerics._
+import org.apache.spark.rdd.RDD
+import org.apache.spark.SparkConf
+import org.apache.spark.SparkContext
+import org.apache.spark.SparkContext._
+import utils._
+
+
+// Trait to be implemented by Kernel functions
+trait KernelGenerator[T] {
```

Comment:
i guess i don't know what the one element equivalent would be. Since a transformer in my head is valid for a single data point, but a kernel isn't really well defined for it?